rust/hyperlane-base/src/types/multisig.rs

```diff
@@ -186,32 +220,40 @@ impl MultisigCheckpointSyncer {
                     let root = signed_checkpoint.value.root;
                     let signed_checkpoints = signed_checkpoints_per_root.entry(root).or_default();
                     signed_checkpoints.push(signed_checkpoint);
+                    cumulative_weight += vw.weight;
 
                     // Count the number of signatures for this signed checkpoint
                     let signature_count = signed_checkpoints.len();
                     debug!(
-                        validator = format!("{:#x}", validator),
+                        validator = format!("{:#x}", vw.validator),
                         index = index,
                         root = format!("{:#x}", root),
                         signature_count = signature_count,
                         "Found signed checkpoint"
                     );
 
-                    // If we've hit a quorum, create a MultisigSignedCheckpoint
-                    if signature_count >= threshold {
+                    // If we've hit a quorum in weight, create a MultisigSignedCheckpoint
+                    if cumulative_weight >= threshold_weight {
+                        // to conform to the onchain ordering of the set by address
+                        signed_checkpoints.sort_by_key(|sc| {
+                            weighted_validators
```

Comment:
This feels like an unnecessarily expensive sorting

`sc.recover()` is pretty expensive and iiuc from the docs https://doc.rust-lang.org/std/vec/struct.Vec.html#method.sort_by_key we'll do this a bunch of times, so if an expensive operation is performed `sort_by_cached_key` is likely to be faster

But regardless this approach feels a bit wasteful, is there a way we can avoid the `sc.recover()` stuff?
Maybe by changing the type of `signed_checkpoints_per_root` to include a property with the intended index of the validator so it's easier to sort later